# DP: Description: Add missing bits in 2.6.19 kernel headers
# DP: Author: Jeff Bailey <jbailey@ubuntu.com>
# DP: Upstream status: Not Submitted.
# DP: Status Details: No response on http://sourceware.org/ml/libc-alpha/2006-10/msg00024.html
# DP: Date: 2006-10-27

--- sysdeps/unix/sysv/linux/ifaddrs.c.old	2006-10-26 21:58:12.000000000 -0400
+++ sysdeps/unix/sysv/linux/ifaddrs.c	2006-10-26 23:06:12.000000000 -0400
@@ -33,9 +33,14 @@
 #include <sysdep.h>
 #include <time.h>
 #include <unistd.h>
+#include <linux/if_addr.h>
 
 #include "netlinkaccess.h"
 
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifinfomsg))
+#define IFA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifaddrmsg))))
+#define IFA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
 
 /* We don't know if we have NETLINK support compiled in in our
    Kernel, so include the old implementation as fallback.  */
--- sysdeps/unix/sysv/linux/check_pf.c.old	2006-10-26 23:04:57.000000000 -0400
+++ sysdeps/unix/sysv/linux/check_pf.c	2006-10-26 23:05:46.000000000 -0400
@@ -30,10 +30,15 @@
 #include <asm/types.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#include <linux/if_addr.h>
 
 #include <not-cancel.h>
 #include <kernel-features.h>
 
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifinfomsg))
+#define IFA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifaddrmsg))))
+#define IFA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
 
 #ifndef IFA_F_TEMPORARY
 # define IFA_F_TEMPORARY IFA_F_SECONDARY
--- sysdeps/unix/sysv/linux/if_index.c.old	2006-10-26 23:38:48.000000000 -0400
+++ sysdeps/unix/sysv/linux/if_index.c	2006-10-26 23:39:42.000000000 -0400
@@ -28,9 +28,14 @@
 #include <sys/ioctl.h>
 #include <bits/libc-lock.h>
 #include <not-cancel.h>
+#include <linux/if_addr.h>
 
 #include "netlinkaccess.h"
 
+#define IFA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifaddrmsg))))
+#define IFA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifaddrmsg))
+#define IFLA_RTA(r)  ((struct rtattr*)(((char*)(r)) + NLMSG_ALIGN(sizeof(struct ifinfomsg))))
+#define IFLA_PAYLOAD(n) NLMSG_PAYLOAD(n,sizeof(struct ifinfomsg))
 
 /* Variable to signal whether SIOCGIFCONF is not available.  */
 # if __ASSUME_SIOCGIFNAME == 0
